app/src/components/Header.module.css

```diff
@@ -1,20 +1,24 @@
+:root {
+  --header-padding-shrink: clamp(0px, calc(var(--y) / 4), 10px);
+  --header-logo-shrink: clamp(0px, calc(var(--y) / 2), 10px);
+  --header-shrink: calc(
+    var(--header-padding-shrink) + var(--header-logo-shrink)
+  );
+}
+
 .header {
   z-index: 2;
   position: sticky;
   top: 0;
-  padding: 20px;
+  margin-bottom: var(--header-shrink);
```

Comment:
CSS var tricks to shrink header on scroll down without messing up scroll and making it jumpy. Add margin to bottom to keep overall space taken up by header the same even as it shrinks in visible size.